src/libraries/System.Drawing.Common/src/System/Drawing/Icon.Windows.cs

```diff
@@ -416,8 +417,11 @@ internal void Draw(Graphics graphics, int x, int y)
         internal void Draw(Graphics graphics, Rectangle targetRect)
         {
             Rectangle copy = targetRect;
-            copy.X += (int)graphics.Transform.OffsetX;
-            copy.Y += (int)graphics.Transform.OffsetY;
+
+            using Matrix transform = graphics.Transform;
```

Comment:
Wow, a property that allocates a new disposable on every access... awesome ;-)

Reply:
I've got a proposal up to help address this, um, suboptimal pattern: https://github.com/dotnet/runtime/issues/47940